cranelift/codegen/src/ir/types.rs

```diff
@@ -79,6 +79,29 @@ impl Type {
         }
     }
 
+    /// Get the (minimum, maximum) values represented by each lane in the type.
+    pub fn bounds(self, signed: bool) -> (i128, i128) {
+        if signed {
+            match self.lane_type() {
+                I8 => (i8::MIN as i128, i8::MAX as i128),
+                I16 => (i16::MIN as i128, i16::MAX as i128),
+                I32 => (i32::MIN as i128, i32::MAX as i128),
+                I64 => (i64::MIN as i128, i64::MAX as i128),
+                I128 => (i128::MIN, i128::MAX),
+                _ => unimplemented!(),
+            }
+        } else {
+            match self.lane_type() {
+                I8 => (u8::MIN as i128, u8::MAX as i128),
+                I16 => (u16::MIN as i128, u16::MAX as i128),
+                I32 => (u32::MIN as i128, u32::MAX as i128),
+                I64 => (u64::MIN as i128, u64::MAX as i128),
+                I128 => (u128::MIN as i128, u128::MAX as i128),
```

Comment:
The `u128::MAX as i128` will overflow, which I guess is fine because I think this is just using the `i128` as a container for the 128 bits, immediately passed into a `Value::int` call. But given that, could you use `u128` instead? IMHO the unsigned types are a bit more idiomatic for "just a bit pattern"-type values.

Reply:
Sure; I'd gone for `i128` because `Value::int` accepts `i128` rather than `u128`.